docker/regtest/docker-compose.yml

```diff
@@ -0,0 +1,153 @@
+        networkactive=1
+        dnsseed=0
+        uacomment=tbkdevbitcoindregtest
+        printpriority=1
+        logtimemicros=1
+        zmqpubrawblock=tcp://0.0.0.0:28332
+        zmqpubrawtx=tcp://0.0.0.0:28333
+        zmqpubhashblock=tcp://0.0.0.0:28334
+        # tor
+        dns=1
+        onion=tor:9050
+        torcontrol=tor:9051
+        # do not automatically create tor hidden service
+        listenonion=0
+        # rpcauth (user=regtest; password=regtest)
+        rpcauth=regtest:20b58677979ad9d3cf4b78b1d6e85e44$$2ec3e1e1c00c7c58d7aff1d4bf96e4a984ea1af5d676d862fd0faa857a1d4d7c
+        # rpcauth (user=joinmarket; password=joinmarket)
+        rpcauth=joinmarket:260b4c5b1fbd09d75a4aabf90226282f$$76e170af088d43a588992cdd5e7bae2242b03c33aa672cccfd1fb75f9281299e
+        # rpcauth (user=nbxplorer; password=nbxplorer)
+        rpcauth=nbxplorer:8953867752ed07ca27bc0b1d7a10fa99$$e46553870d4a2c59753bfe453797be070ab4dbd5fdfc2fecdb7862ea0ae6b127
+    expose:
+      - "43782" # RPC
+      - "39388" # P2P
+      - "28332" # ZMQ
+      - "28333" # ZMQ
+      - "28334" # ZMQ
+    volumes:
+      - "bitcoin_datadir:/data"
+      - "bitcoin_wallet_datadir:/walletdata"
+      - "tor_datadir:/home/tor/.tor"
+    depends_on:
+      - tor
+
+  nbxplorer:
+    container_name: jm_regtest_nbxplorer
+    restart: unless-stopped
+    image: nicolasdorier/nbxplorer:2.2.18
+    environment:
+      NBXPLORER_NOAUTH: 1
+      NBXPLORER_CHAINS: "btc"
+      NBXPLORER_BTCRPCURL: http://bitcoind:43782/
+      NBXPLORER_BTCRPCUSER: nbxplorer
+      NBXPLORER_BTCRPCPASSWORD: nbxplorer
+      NBXPLORER_BTCNODEENDPOINT: bitcoind:39388
+      NBXPLORER_NETWORK: ${NBITCOIN_NETWORK:-regtest}
+      NBXPLORER_BIND: 0.0.0.0:32838
+      NBXPLORER_TRIMEVENTS: 10000
+      NBXPLORER_SIGNALFILESDIR: /datadir
+    expose:
+      - "32838"
+    volumes:
+      - "nbxplorer_datadir:/datadir"
+      - "bitcoin_datadir:/root/.bitcoin"
+    depends_on:
+      - bitcoind
+      - tor
+
+  miniircd:
+    container_name: jm_regtest_miniircd
+    restart: unless-stopped
+    image: daxxog/miniircd:latest
+    expose:
+      - 6667
+
+  joinmarket:
+    container_name: jm_regtest_joinmarket
+    build:
+      context: ./dockerfile-deps/joinmarket
+      dockerfile: Dockerfile
+    restart: unless-stopped
+    environment:
+      READY_FILE: /root/.nbxplorer/btc_fully_synched
+      ENSURE_WALLET: 1
+      jm_gaplimit: 2000
+      #jm_network: ${NBITCOIN_NETWORK:-regtest}
+      jm_blockchain_source: regtest
+      jm_network: testnet
+      jm_rpc_host: bitcoind
+      jm_rpc_port: 43782
+      jm_rpc_user: joinmarket
+      jm_rpc_password: joinmarket
+      jm_tor_control_host: tor
+      jm_tor_control_port: 9051
+      jm_onion_socks5_host: tor
+      jm_onion_socks5_port: 9050
+      jm_socks5_host: tor
+      jm_socks5_port: 9050
+    expose:
+      - 8080 # payjoin server
+      - 62601 # obwatch
+      - 27183 # joinmarketd
+      - 28183 # jmwalletd
+    volumes:
```

Comment:
```suggestion
    ports:
      - "8080:8080"
      - "62601:62601"
      - "27183:27183"
      - "28183:28183"
    volumes:
```

Reply:
Nice. Is it okay to just map 28183 and 62601 for now? Don't want to mess with peoples local setup for 8080.

Reply:
Yes, good point!